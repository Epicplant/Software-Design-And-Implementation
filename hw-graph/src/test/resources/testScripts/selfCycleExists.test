# Tests correct functionality with a self-loop

# Creates a new graph
CreateGraph newGraph

# Creates the cycles first node
AddNode newGraph n1

# Creates a self-loop for n1 with weight 0
AddEdge newGraph n1 n1 0

# Lists all the children of node n1
ListChildren newGraph n1

# Lists all the nodes in the newGraph
ListNodes newGraph