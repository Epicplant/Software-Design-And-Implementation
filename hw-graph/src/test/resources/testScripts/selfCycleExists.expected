# Tests correct functionality with a self-loop

# Creates a new graph
created graph newGraph

# Creates the cycles first node
added node n1 to newGraph

# Creates a self-loop for n1 with weight 0
added edge 0 from n1 to n1 in newGraph

# Lists all the children of node n1
the children of n1 in newGraph are: n1(0)

# Lists all the nodes in the newGraph
newGraph contains: n1